packages/babel-plugin-transform-es2015-modules-commonjs/src/index.js

```diff
@@ -439,14 +456,16 @@ export default function () {
           }
 
           // add __esModule declaration if this file has any exports
-          if (hasExports && !strict) {
```

Comment:
I haven't been able to understand the purpose of this check (according to git blame, it has been added in the single big commuit between 5.X -> 6.X). It doesn't match the comment above, which is weird.